core/new-gui/src/app/workspace/service/workflow-graph/model/workflow-action.service.ts

```diff
@@ -140,6 +193,8 @@ export class WorkflowActionService {
           undo: () => this.deleteLinkWithIDInternal(link.linkID),
           redo: () => this.addLinkInternal(link),
         };
+        const commandMessage: CommandMessage = { action: "addLink", parameters: [link], type: "execute" };
```

Comment:
I think this function can direclty call `addLink` inside `subscribe`, can you please try to refactor it so we have less redundancy?